opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -1466,6 +1466,43 @@ func resourceOpennebulaVirtualMachineUpdate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if d.HasChange("cpu") || d.HasChange("vcpu") || d.HasChange("memory") {
+
+		vmState, _, _ := vmInfos.State()
+		vmRequireShutdown := vmState != vm.Poweroff && vmState != vm.Undeployed
+		if vmRequireShutdown {
+			err = vmc.Poweroff()
+			if err != nil {
+				return err
+			}
+
+			timeout := d.Get("timeout").(int)
+			_, err = waitForVMState(vmc, timeout, "POWEROFF")
+			if err != nil {
+				return fmt.Errorf(
+					"waiting for virtual machine (ID:%d) to be in state %s: %s", vmc.ID, "POWEROFF", err)
+			}
+		}
+
+		resizeTpl := dyn.NewTemplate()
+		resizeTpl.AddPair("CPU", d.Get("cpu").(float64))
+		resizeTpl.AddPair("VCPU", d.Get("vcpu").(int))
+		resizeTpl.AddPair("MEMORY", d.Get("memory").(int))
+
+		err = vmc.Resize(resizeTpl.String(), true)
+		if err != nil {
+			return err
+		}
+
+		if vmRequireShutdown {
+			err = vmc.Resume()
+			if err != nil {
+				return err
+			}
+		}
+		log.Printf("[INFO] Successfully resized VM %s\n", vmInfos.Name)
```

Comment:
you need to wait here Resume is OK. It is possible the VM will not reach a RUNNING state here if it does not boot.

Reply:
@jaypif  fixed it